packages/app/components/Queue/QueueListSharedComponents.tsx

```diff
@@ -22,21 +25,88 @@ const TimeText = styled.div`
   margin-left: 12px;
 `;
 
-const NotesText = styled.div`
+export const NotesText = styled.div`
   font-size: 16px;
   color: #5f6b79;
 `;
 
+const AvatarWithMargin = styled(Avatar)`
+  margin-right: 10px;
+`;
+
+const NotesInput = styled(Input)`
+  border-radius: 6px;
+  border: 1px solid #b8c4ce;
+`;
+
 interface QueueListHeaderProps {
-  queue: QueuePartial;
+  queueId: number;
+  isTA: boolean;
 }
 
 export default function QueueListHeader({
-  queue,
+  queueId,
+  isTA,
 }: QueueListHeaderProps): ReactElement {
+  const { queue, queuesError, mutateQueue } = useQueue(queueId);
+  const [queueSettingsModal, setQueueSettingsModal] = useState(false);
+  const [oldNotes, setOldNotes] = useState(queue?.notes);
+  const [updatedNotes, setUpdatedNotes] = useState(queue?.notes);
+
+  const updateQueueNotes = async () => {
+    await API.queues.updateNotes(queueId, updatedNotes);
+    setOldNotes(updatedNotes);
+    mutateQueue();
+  };
+
+  const toggleAllowQuestions = async () => {
+    await API.queues.toggleQueue(queueId);
+    mutateQueue();
+  };
+
   return (
     <Container>
       <QueueTitle>{queue?.room}</QueueTitle>
+      {isTA && (
+        <Tooltip title="Cool admin things that TAs like you can do yeah">
+          <SettingOutlined
+            style={{ fontSize: 20, paddingLeft: 24 }}
+            onClick={() => setQueueSettingsModal(true)}
+          />
+        </Tooltip>
+      )}
+      <Modal
+        title="LOL Stanley you're gonna have to figure this one out"
+        visible={queueSettingsModal}
+        onCancel={() => {
+          setUpdatedNotes(oldNotes);
+          setQueueSettingsModal(false);
+        }}
+        onOk={() => {
+          setOldNotes(updatedNotes);
+          setQueueSettingsModal(false);
+        }}
```

Comment:
Looks like `updateQueueNotes` need to be called somewhere in here
```suggestion
        onOk={() => {
          setOldNotes(updatedNotes);
          updateQueueNotes();
          setQueueSettingsModal(false);
        }}
```